components/file_preview_modal/image_preview.tsx

```diff
@@ -0,0 +1,225 @@
+
+    const touch = useRef({x: 0, y: 0});
+    const canvasBorder = useRef({w: 0, h: 0});
+    const isMouseDown = useRef(false);
+    const containerRef = useRef(null);
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+
+    const isExternalFile = !fileInfo.id;
+
+    let fileUrl;
+    let previewUrl: string;
+    if (isExternalFile) {
+        fileUrl = fileInfo.link;
+        previewUrl = fileInfo.link;
+    } else {
+        fileUrl = getFileDownloadUrl(fileInfo.id);
+        previewUrl = fileInfo.has_preview_image ? getFilePreviewUrl(fileInfo.id) : fileUrl;
+    }
+
+    const clampOffset = (x: number, y: number) => {
+        // Clamps the offset to something that is inside canvas or window depending on zoom level
+        const {w, h} = canvasBorder.current;
+        const {horizontal, vertical} = isFullscreen;
+
+        if (zoom <= maxCanvasZoom) {
+            return {xPos: 0, yPos: 0};
+        }
+
+        return {
+            xPos: horizontal ? clamp(x, w, -w) : 0,
+            yPos: vertical ? clamp(y, h, -h) : 0,
+        };
+    };
+
+    const handleWheel = (event: React.WheelEvent) => {
+        event.persist();
+        const {deltaY} = event;
+        if (!dragging) {
+            zoom = clamp(zoom + (deltaY * SCROLL_SENSITIVITY * -1), minZoom, MAX_ZOOM);
+            setToolbarZoom(zoom === minZoom ? 'A' : zoom);
+        }
+    };
+
+    const handleMouseMove = (event: React.MouseEvent) => {
+        if (!dragging) {
+            return;
+        }
+
+        const {x, y} = touch.current;
+        const {clientX, clientY} = event;
+        const {xPos, yPos} = clampOffset(offset.x + (x - clientX), offset.y + (y - clientY));
+        setOffset({x: xPos, y: yPos});
+        touch.current = {x: clientX, y: clientY};
+    };
+
+    const handleMouseDown = (event: React.MouseEvent) => {
+        event.preventDefault();
+        const {clientX, clientY} = event;
+        touch.current = {x: clientX, y: clientY};
+        isMouseDown.current = true;
+        setDragging(true);
+    };
+
+    const handleMouseUp = () => {
+        isMouseDown.current = false;
+        setDragging(false);
+    };
+
+    // Stop dragging if mouse left canvas
+    const handleMouseLeave = () => setDragging(false);
+
+    // Resume dragging if mouse stays clicked
+    const handleMouseEnter = () => setDragging(isMouseDown.current);
+
+    // Initialize canvas
+    useEffect(() => {
+        // Global mouseup event, otherwise canvas can stay stuck on mouse when leaving canvas while dragging
+        window.addEventListener('mouseup', handleMouseUp);
+
+        background.src = previewUrl;
+        if (canvasRef.current) {
+            background.onload = () => {
+                // Initialize with the zoom at minimum.
+                setIsReady(true);
+            };
+        }
+
+        return () => {
+            window.removeEventListener('mouseup', handleMouseUp);
+        };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
is this eslint error or warning? if warning then we dont have to include the disable comment